sdk/iOS/test/MSManagedObjectObserverTests.m

```diff
@@ -0,0 +1,176 @@
+//
+//  MSManagedObjectObserverTests.m
+//  WindowsAzureMobileServices
+//
+//  Created by Damien Pontifex on 13/03/2015.
+//  Copyright (c) 2015 Windows Azure. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "MSCoreDataStore.h"
+#import "MSCoreDataStore+TestHelper.h"
+#import "MSJSONSerializer.h"
+#import "TodoItem.h"
+#import "MSManagedObjectObserver.h"
+#import "MSTableOperation.h"
+
+@interface MSManagedObjectObserverTests : XCTestCase
+@property (nonatomic, strong) MSClient *client;
+@property (nonatomic, strong) MSCoreDataStore *store;
+@property (nonatomic, strong) NSManagedObjectContext *context;
+@property (nonatomic, strong) MSManagedObjectObserver *observer;
+
+@property (nonatomic, strong) TodoItem *item;
+@end
+
+@implementation MSManagedObjectObserverTests
+
+/// Helper method so we can start observing at the point we want to rather than initialized during setup
+- (void)startObservingContextWithObservationCompletion:(MSManagedObjectObserverCompletionBlock)completionBlock
+{
+	self.observer = [[MSManagedObjectObserver alloc] initWithClient:self.client];
+	self.observer.observerActionCompleted = completionBlock;
+}
+
+- (void)insertTestItem
+{
+	self.item = [NSEntityDescription insertNewObjectForEntityForName:@"TodoItem" inManagedObjectContext:self.context];
+	
+	self.item.text = @"Test item";
+	self.item.id = @"ABC";
+	
+	[self.context save:nil];
+}
+
+- (void)setUp
+{
+    [super setUp];
+	
+	self.context = [MSCoreDataStore inMemoryManagedObjectContext];
+	self.store = [[MSCoreDataStore alloc] initWithManagedObjectContext:self.context];
+	
+	self.client = [[MSClient alloc] initWithApplicationURL:nil applicationKey:nil];
+	self.client.syncContext = [[MSSyncContext alloc] initWithDelegate:nil dataSource:self.store callback:nil];
+}
+
+- (void)tearDown
+{
+	self.store = nil;
+	
+    [super tearDown];
+}
+
+- (void)testObservingInsertOperation
+{
+	XCTestExpectation *expectation = [self expectationWithDescription:@"Table operation observed"];
```

Comment:
I know. Allowed me to remove a whole heap of dispatch semaphore and dispatch group waits in another project I'm working on, so picked it up quite early :)